Sources/Configurations/CreditCardConfiguration.swift

```diff
@@ -0,0 +1,135 @@
+        self.rawValue = rawValue
+    }
+    
+    static let americanExpress    = CreditCardType(rawValue: 1 << 0)
+    static let dinersClub         = CreditCardType(rawValue: 1 << 1)
+    static let discover           = CreditCardType(rawValue: 1 << 2)
+    static let jcb                = CreditCardType(rawValue: 1 << 3)
+    static let maestro            = CreditCardType(rawValue: 1 << 4)
+    static let mastercard         = CreditCardType(rawValue: 1 << 5)
+    static let visa               = CreditCardType(rawValue: 1 << 6)
+    
+    static let all: CreditCardType = [.americanExpress, .dinersClub, .discover, .jcb, .maestro, .mastercard, .visa]
+    static let allArray: [CreditCardType] = [.americanExpress, .dinersClub, .discover, .jcb, .maestro, .mastercard, .visa]
+    
+    /**
+     Literal description for set of credit cards, separated with commas and single space.
+     */
+    public var description: String {
+        
+        var result = ""
+        
+        add(description: "American Express", forCardType: CreditCardType.americanExpress, toResultString: &result)
+        add(description: "Diners Club", forCardType: CreditCardType.dinersClub, toResultString: &result)
+        add(description: "Discover", forCardType: CreditCardType.discover, toResultString: &result)
+        add(description: "JCB", forCardType: CreditCardType.jcb, toResultString: &result)
+        add(description: "Maestro", forCardType: CreditCardType.maestro, toResultString: &result)
+        add(description: "Master Card", forCardType: CreditCardType.mastercard, toResultString: &result)
+        add(description: "VISA", forCardType: CreditCardType.visa, toResultString: &result)
+        
+        return result
+    }
+    
+    /**
+     Combined regular expression string for set of credit cards.
+     */
+    public var regex: String {
+        var result = ""
+        self.insertCard(regexString: &result, cardType: .americanExpress)
+        self.insertCard(regexString: &result, cardType: .dinersClub)
+        self.insertCard(regexString: &result, cardType: .discover)
+        self.insertCard(regexString: &result, cardType: .jcb)
+        self.insertCard(regexString: &result, cardType: .maestro)
+        self.insertCard(regexString: &result, cardType: .mastercard)
+        self.insertCard(regexString: &result, cardType: .visa)
+        
+        return result
+    }
+    
+    /**
+     Adds commas and spaces between credit card descriptions.
+     */
+    private func add(description descriptionString: String, forCardType cardType: CreditCardType, toResultString resultString: inout String) {
+        if self.contains(cardType) {
+            resultString += ((!resultString.isEmpty) ? ", " : "")
+            resultString += descriptionString
+        }
+    }
+    
+    /**
+     Regular expression values for credit card types.
+     */
+    private var cardRegex: String? {
+        switch self {
+        case CreditCardType.americanExpress:
```

Comment:
I don't suppose we have any links to documentation or reference we can add here for the regex e.g. Luhn algorithm?